apps/total-typescript/src/pages/api/inngest.ts

```diff
@@ -1,24 +1,100 @@
 import {serve} from 'inngest/next'
 import {convertkitSurveyAnswered, inngest} from '@skillrecordings/inngest'
+import {ChatOpenAI} from 'langchain/chat_models'
+import {
+  ChatPromptTemplate,
+  HumanMessagePromptTemplate,
+  SystemMessagePromptTemplate,
+} from 'langchain/prompts'
 
-type HelloWorld = {
-  name: 'test/hello.world'
+import {ConversationChain} from 'langchain/chains'
+
+type SummaryInputEvent = {
+  name: 'api/chat.submitted'
   data: {
-    name: string
-    email: string
+    requestId: string
+    input: string
   }
 }
 type Events = {
-  'test/hello.world': HelloWorld
+  'api/chat.submitted': SummaryInputEvent
 }
 
-const helloWorld = inngest.createFunction(
-  {name: 'Hello Worldzzzz'},
-  {event: 'test/hello.world'},
+const summarizeInput = inngest.createFunction(
+  {name: 'Summarize chat and documents'},
+  {event: 'api/chat.submitted'},
   async ({event, step}) => {
-    await step.sleep('1s')
-    return {event, body: 'Hello, World!'}
+    const output = await step.run('Summarize input', async () => {
```

Comment:
with inngest, each of these steps in run as individual serverless processes in sequence (or parallel)

there are limits to this approach, but also ways to mitigate

https://www.inngest.com/blog/vercel-long-running-background-functions
https://www.inngest.com/blog/running-chained-llms-typescript-in-production